crates/notion-core/src/tool/mod.rs

```diff
@@ -69,6 +72,28 @@ impl ToolSpec {
         }
         Ok(())
     }
+
+    pub fn try_from_str(tool_spec: &str) -> Fallible<Self> {
+        let re = Regex::new("(?P<name>@?[-/_A-z]+)(@(?P<version>.+))?").expect("regex is valid");
```

Comment:
I just realized: I need to double-check this regex against the actual spec for package names!

Reply:
Ah, didn't see your edit while I was looking at this. Looks like that should fix the issues I found.

Reply:
I ended up solving this by using the relatively loose regex I had here, just with a `^` to make sure it only matches the start of the string and tweaked slightly to match the regex used by npm, and then implementing more rigorous testing in a way we can later upstream in the new `validate-npm-package-name` crate I added (and which we can extract to its own crate at will).